Support specifying multiple templates names in delete component template api

martijnvg opened the pull request:
Add support to delete component templates api to specify multiple template
names separated by a comma.

Change the cleanup template logic for rest tests to remove all component templates via a single delete component template request.
This to optimize the cleanup logic. After each rest test we delete all templates. So deleting templates this via a single api call
(and thus single cluster state update) saves a lot of time considering the number of rest tests.

Relates to #69973

martijnvg committed:
Support specifying multiple templates names in delete component template api

Add support to delete component templates api to specify multiple template
names separated by a comma.

Change the cleanup template logic for rest tests to remove all component templates via a single delete component template request.
This to optimize the cleanup logic. After each rest test we delete all templates. So deleting templates this via a single api call
(and thus single cluster state update) saves a lot of time considering the number of rest tests.

Relates to #69973

elasticmachine commented:
Pinging @elastic/es-core-features (Team:Core/Features)

dakrone commented on server/src/main/java/org/elasticsearch/action/admin/indices/template/delete/DeleteComponentTemplateAction.java:
```diff
@@ -30,50 +34,48 @@ private DeleteComponentTemplateAction() {
-        public Request name(String name) {
-            this.name = name;
-            return this;
+        public Request(String... name) {
+            this.names = Objects.requireNonNull(name, "component templates to delete must not be null");
```
Super minor nit
```suggestion
        public Request(String... names) {
            this.names = Objects.requireNonNull(names, "component templates to delete must not be null");
```

dakrone reviewed (approved):
LGTM, I left one super minor comment

martijnvg committed:
fixed typo

Co-authored-by: Lee Hinman <dakrone@users.noreply.github.com>

martijnvg committed:
fixed yaml tests

martijnvg committed:
fixed yaml tests again and enforced that component and composable index templates are
returned in a consistent order in the respective get APIs.

martijnvg committed:
fixed checkstyle violation

martijnvg committed:
undo changes to get component and composable index templates apis.

martijnvg committed:
don't try to delete v2 templates if all nodes are below version 7.7.0

martijnvg committed:
checkstyle

martijnvg committed:
Older versions don't support component / composable index templates
and/or data streams. Yet the test base class tries to remove objects
after each test, which adds a significant number of lines to the
log files (which slows the tests down). The ESRestTestCase will
now check whether all nodes have a specific version and then decide
whether data streams and component / composable index templates will
be deleted.

Also ensured that the logstash-index-template and security-index-template
aren't deleted between tests, these templates are builtin templates that
ES will install if missing. So if tests remove these templates between tests
then ES will add these template back almost immediately. These causes
many log lines and a lot of cluster state updates, which slow tests down.

martijnvg committed:
Merge remote-tracking branch 'es/master' into delete_component_templates_comma_seperated

martijnvg committed:
iter
